pina/geometry/simplex.py

```diff
@@ -0,0 +1,246 @@
+import torch
+from .location import Location
+from pina.geometry import CartesianDomain
+from pina import LabelTensor
+from ..utils import check_consistency
+
+
+class SimplexDomain(Location):
+    """PINA implementation of a Simplex."""
+
+    def __init__(self, simplex_matrix, sample_surface=False):
+        """
+        :param simplex_matrix: A matrix of LabelTensor objects representing
+            a vertex of the simplex (a tensor), and the coordinates of the
+            point (a list of labels).
+        :type simplex_matrix: list[LabelTensor]
+        :param sample_surface: A variable for choosing sample strategies. If
+            `sample_surface=True` only samples on the Simplex surface
+            frontier are taken. If `sample_surface=False`, no such criteria
+            is followed.
+        :type sample_surface: bool
+
+        .. warning::
+            Sampling for dimensions greater or equal to 10 could result
+            in a shrinking of the simplex, which degrades the quality
+            of the samples. For dimensions higher than 10, other algorithms
+            for sampling should be used.
+
+        :Example:
+
+            >>> spatial_domain = SimplexDomain(
+                    [
+                        LabelTensor(torch.tensor([[0, 0]]), labels=["x", "y"]),
+                        LabelTensor(torch.tensor([[1, 1]]), labels=["x", "y"]),
+                        LabelTensor(torch.tensor([[0, 2]]), labels=["x", "y"]),
+                    ], sample_surface = True
+                )
+        """
+
+        # check consistency of sample_surface
+        check_consistency(sample_surface, bool)
+        self._sample_surface = sample_surface
+
+        # check consistency of simplex_dict
+        if not isinstance(simplex_matrix, list):
+            raise ValueError(f"{type(vertex).__name__} must be {list}.")
+        try:
+            _labels = simplex_matrix[0].labels
+        except:
+            raise ValueError(
+                f"{type(simplex_matrix[0]).__name__} must be {LabelTensor}."
+            )
+
+        for vertex in simplex_matrix:
+            if not isinstance(vertex, LabelTensor):
```

Comment:
this can be removed, check only once that you have label tensors with the same labels.